tests/ClientTest.php

```diff
@@ -38,65 +31,53 @@ public function testConstructorInitializesTransactionStack()
-    {
-        $client = ClientBuilder::create()->getClient();
-
-        $this->assertEmpty($client->getTransactionStack());
-    }
-
-    public function testGetTransactionStack()
-    {
-        $client = ClientBuilder::create()->getClient();
-
-        $this->assertInstanceOf(TransactionStack::class, $client->getTransactionStack());
-    }
+        /** @var TransportInterface|\PHPUnit_Framework_MockObject_MockObject $transport */
+        $transport = $this->createMock(TransportInterface::class);
 
-    public function testAddMiddleware()
-    {
-        $middleware = function () {};
+        /** @var Client|\PHPUnit_Framework_MockObject_MockObject $client */
+        $client = $this->getMockBuilder(Client::class)
+            ->setConstructorArgs([new Options(), $transport])
+            ->setMethodsExcept(['captureMessage', 'prepareEvent', 'captureEvent', 'getOptions'])
+            ->getMock();
 
-        /** @var MiddlewareStack|\PHPUnit_Framework_MockObject_MockObject $middlewareStack */
-        $middlewareStack = $this->createMock(MiddlewareStack::class);
-        $middlewareStack->expects($this->once())
-            ->method('addMiddleware')
-            ->with($middleware, -10);
+        $client->expects($this->once())
+            ->method('send')
+            ->with($this->callback(function ($event) {
+                /* @var Event $event*/
+                $this->assertInstanceOf(Event::class, $event);
+                $this->assertEquals('/foo', $event->getTransaction());
 
-        $client = ClientBuilder::create()->getClient();
+                return true;
+            }));
 
-        $reflectionProperty = new \ReflectionProperty($client, 'middlewareStack');
-        $reflectionProperty->setAccessible(true);
-        $reflectionProperty->setValue($client, $middlewareStack);
-        $reflectionProperty->setAccessible(false);
+        $client->captureMessage('test');
 
-        $client->addMiddleware($middleware, -10);
+        unset($_SERVER['PATH_INFO']);
+        unset($_SERVER['REQUEST_METHOD']);
     }
 
-    public function testRemoveMiddleware()
+    public function testConstructorInitializesTransactionStackInCli()
     {
-        $middleware = function () {};
+        /** @var TransportInterface|\PHPUnit_Framework_MockObject_MockObject $transport */
+        $transport = $this->createMock(TransportInterface::class);
 
-        /** @var MiddlewareStack|\PHPUnit_Framework_MockObject_MockObject $middlewareStack */
-        $middlewareStack = $this->createMock(MiddlewareStack::class);
-        $middlewareStack->expects($this->once())
-            ->method('removeMiddleware')
-            ->with($middleware);
+        /** @var Client|\PHPUnit_Framework_MockObject_MockObject $client */
+        $client = $this->getMockBuilder(Client::class)
+            ->setConstructorArgs([new Options(), $transport])
+            ->setMethodsExcept(['captureMessage', 'prepareEvent', 'captureEvent', 'getOptions'])
```

Comment:
A partial mock is a bad way to proceed. You should set expectation on the transport dependency instead.